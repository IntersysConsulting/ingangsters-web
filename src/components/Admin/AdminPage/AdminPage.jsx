import React from "react";
import AdminTabBar from "../AdminTabBar/AdminTabBar";
import { connect } from "react-redux";
import AdminNavBar from "../../NavBars/AdminNavBar/AdminNavBar";
import AdminProducts from "../Products/index";

const AdminPage = ({ adminOption }) => {
  return (
    <div>
      <AdminNavBar />
      <AdminTabBar />
      <br />
      <br />

      {adminOption === "Users" ? (
        <center></center>
      ) : adminOption === "Products" ? (
        <center>
          <AdminProducts></AdminProducts>
        </center>
      ) : (
        <center>
          <p>Orders</p>
        </center>
      )}
    </div>
  );
};

function mapStateToProps(state) {
  return {
    adminOption: state.adminTabBar.activeButton
  };
}

function mapDispatchToProps(dispatch) {
  return {};
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage);
