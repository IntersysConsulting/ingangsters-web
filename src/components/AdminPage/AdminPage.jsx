import React from "react";
import AdminTabBar from "../AdminTabBar/AdminTabBar";
import Shadow from "../Shadow/Shadow";
import { connect } from "react-redux";
import AdminNavBar from "../AdminNavBar/AdminNavBar";
import AdminProducts from "../Admin/Products/index";

const AdminPage = ({ adminOption }) => {
  return (
    <div>
      <AdminNavBar />
      <Shadow />
      <AdminTabBar />
      <br />
      <br />

      {adminOption == "Users" ? (
        <center>
          <p>Users</p>
        </center>
      ) : adminOption == "Products" ? (
        <center>
          <p>Products</p>
          <AdminProducts></AdminProducts>
        </center>
      ) : (
        <center>
          <p>Orders</p>
        </center>
      )}
    </div>
  );
};

function mapStateToProps(state) {
  return {
    adminOption: state.adminTabBar.activeButton
  };
}

function mapDispatchToProps(dispatch) {
  return {};
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage);
