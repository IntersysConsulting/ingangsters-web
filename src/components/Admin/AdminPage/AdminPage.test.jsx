import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter, Provider } from "react-router-dom";
import AdminPage from "./AdminPage";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";

describe("Admin Page", () => {
  const initialState = {
    adminTabBar: {
      activeButton: "Users"
    }
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const actions = [];
  const store = mockStore(initialState, actions);

  it("renders correctly", () => {
    const wrapper = shallow(<AdminPage store={store} dispatch={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
