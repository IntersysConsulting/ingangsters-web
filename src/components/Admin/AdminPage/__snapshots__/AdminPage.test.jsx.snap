// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <AdminPage
    adminOption="Users"
    dispatch={null}
    fetchProducts={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
